src/gui/curses/gui-curses-main.c

```diff
@@ -319,6 +319,18 @@ gui_main_handle_quit_signals ()
     weechat_quit_signal = 0;
 }
 
+/*
+ * Callback for signals received that will make WeeChat reload.
+ */
+
+void
+gui_main_handle_reload_signal ()
+{
+    log_printf ("Signal SIGHUP received, reloading WeeChat...");
```

Comment:
It would be more accurate to say "reloading WeeChat configuration" (WeeChat itself is not reloaded).